lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx

```diff
@@ -141,66 +146,73 @@ describe('NxSegmentedButton', function() {
     expect(onToggleOpen).not.toHaveBeenCalled();
 
     act(() => {
-      component!.find('button.nx-segmented-btn__dropdown-btn').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
+      component!.find('button.nx-segmented-btn__dropdown-btn').at(0)
+          .getDOMNode().dispatchEvent(new MouseEvent('click', {
+            bubbles: true
+          }));
     });
     component!.update();
     expect(onToggleOpen).toHaveBeenCalledTimes(1);
   });
 
   it('does not call the onToggleOpen prop when the component is disabled', function() {
     const onToggleOpen = jest.fn(),
-        component = getShallow({ onToggleOpen, disabled: true });
+        component = getMounted({ onToggleOpen, disabled: true });
 
     expect(onToggleOpen).not.toHaveBeenCalled();
 
-    component.find('.nx-segmented-btn__main-btn').simulate('click');
+    component.find('.nx-segmented-btn__main-btn').at(0).simulate('click');
     expect(onToggleOpen).not.toHaveBeenCalled();
 
-    component.find('.nx-segmented-btn__dropdown-btn').simulate('click');
-    expect(onToggleOpen).toHaveBeenCalled();
+    component.find('.nx-segmented-btn__dropdown-btn').at(0).simulate('click');
+    expect(onToggleOpen).not.toHaveBeenCalled();
```

Comment:
I added a `not` here but not 100% sure if it makes sense; we don't want this to be called when the component is disabled right?

Reply:
> I added a not here but not 100% sure if it makes sense; we don't want this to be called when the component is disabled right?

Hmm that seems to be what the documentation says, and also how it behaves IRL on main.  This leads me to believe that this test is bad.  I think the not-calling-things behavior IRL is based on the fact that a `disabled` button doesn't fire click events in the first place.  But this test is simulating them as being fired, which doesn't accurately reproduce that scenario.  We already have the test below that the buttons are getting the native `disabled` attr, which is probably all we can really do in a unit test.  If we wanted to be extra sure, we could write a functional test for it.

Is this test passing with your change here? If so I don't understand why.

Reply:
> > I added a not here but not 100% sure if it makes sense; we don't want this to be called when the component is disabled right?
> 
> Hmm that seems to be what the documentation says, and also how it behaves IRL on main. This leads me to believe that this test is bad. I think the not-calling-things behavior IRL is based on the fact that a `disabled` button doesn't fire click events in the first place. But this test is simulating them as being fired, which doesn't accurately reproduce that scenario. We already have the test below that the buttons are getting the native `disabled` attr, which is probably all we can really do in a unit test. If we wanted to be extra sure, we could write a functional test for it.
> 
> Is this test passing with your change here? If so I don't understand why.

Yeah this test is passing strangely (with the change here) and would fail if I remove the `not` 🤔

Reply:
> Yeah this test is passing strangely (with the change here) and would fail if I remove the not thinking

Well, regardless I think the test is invalid and should be removed.  It's testing what happens when a disabled dropdown receives a click event, and that's just not something that happens.

Reply:
> > Yeah this test is passing strangely (with the change here) and would fail if I remove the not thinking
> 
> Well, regardless I think the test is invalid and should be removed. It's testing what happens when a disabled dropdown receives a click event, and that's just not something that happens.

Removed: 8bd59c0065da56673b1f11d0db45bad388ccbdea